db/model/collectorgroup.js

```diff
@@ -128,35 +141,20 @@ module.exports = function collectorgroup(seq, dataTypes) {
       }));
   }; // isWritableBy
 
-  CollectorGroup.createCollectorGroup = function (requestBody) {
-    let collectors;
-
-    return Promise.resolve()
-      .then(() => {
-        if (!(requestBody.collectors && requestBody.collectors.length > 0)) {
-          return [];
-        }
-
-        return collectorUtils
-          .validate(seq, requestBody.collectors)
-          .then((validCollectors) => {
-            const alreadyAssigned = validCollectors
-              .filter((collector) => collector.collectorGroupId);
-
-            if (alreadyAssigned && alreadyAssigned.length > 0) {
-              const names = alreadyAssigned.map((c) => c.name);
-              const msg = 'Collector ' + names +
-                ' already assigned to a different group';
-              throw new Error(msg);
-            }
-
-            collectors = validCollectors;
-          });
-      })
-      .then(() => CollectorGroup.create(requestBody))
-      .then((collectorGroup) => collectorGroup.setCollectors(collectors))
-      .then((collectorGroup) => collectorGroup.reload());
-  };
+  /**
+   * Add the named collectors to this collector group. Reject if any of the
+   * named collectors is already assigned to this group, or to a different
+   * group. If array is empty, remove all collectors from this collector group.
+   *
+   * @param {Array<String>} arr - array of collector names
+   * @returns {Promise<any | never>}
+   */
+  CollectorGroup.prototype.addCollectorsToGroup = function (arr) {
+    return collectorUtils.validate(seq, arr)
+      .then(collectorUtils.alreadyAssigned)
+      .then((collectors) => this.setCollectors(collectors))
```

Comment:
Is this supposed to add to existing collectors? Should this be addCollectors instead? Test for this use case?

Reply:
Oh good point--thx!!!